### PR TITLE
Export Consent Forms (Markdown)

### PR DESCRIPTION
**Pull request by philippzagar**
# *Export Consent Forms (Markdown)*

## :recycle: Current situation & Problem
Until now, users could view and sign a consent document in the Spezi template application using the `SpeziOnboarding` package. However, the app did not offer any provision to save or export the signed consent document.
In the medical domain, the consent form collection is especially important in order to serve as legal cover.
See https://github.com/StanfordSpezi/SpeziOnboarding/issues/7.

Utilizes: https://github.com/StanfordSpezi/SpeziFirebase/pull/17

## :gear: Release Notes 
- The template application now utilizes the `OnboardingConsentView` of `SpeziOnboarding`, providing an export and share functionality of the signed consent form as a PDF.

## :books: Documentation
Incorporated where necessary.
Associated with https://github.com/StanfordSpezi/SpeziOnboarding/pull/26

## :white_check_mark: Testing
--

## :pencil: Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

**Comment by philippzagar** [comment]
@PSchmiedmayer PR ready for review, keep in mind that the `feat/export-consent-forms` branch of `SpeziOnboarding` is currently set as a dependency so that you are able to test locally / Pipeline succeeds. As soon as https://github.com/StanfordSpezi/SpeziOnboarding/pull/26 is merged and tagged, I'll update the dependency to a tagged version.

**Comment by philippzagar** [comment]
@PSchmiedmayer Seems like the pipeline is running into some issues:
```
Error: Encountered a fatal error while running "/Users/runner/hostedtoolcache/CodeQL/2.14.6/x64/codeql/codeql database finalize --finalize-dataset --threads=4 --ram=13004
/Users/runner/work/SpeziTemplateApplication/SpeziTemplateApplication/.codeql/swift". 
Exit code was 32 and error was: CodeQL detected code written in Swift but could not process any of it. Review our troubleshooting guide at https://gh.io/troubleshooting-code-scanning/no-source-code-seen-during-build.
```

**Comment by PSchmiedmayer** [comment]
@philippzagar Thank you for the PR. I will give https://github.com/StanfordSpezi/SpeziOnboarding/pull/26 a review and then we can merge this one here. Also took a look at the build errors in #39 & #41, that should also fix the CI in this PR 👍

**Comment by philippzagar** [comment]
@PSchmiedmayer Ready for review after you reviewed https://github.com/StanfordSpezi/SpeziOnboarding/pull/26 :)

File: TemplateApplication/TemplateApplicationStandard.swift
Comment (PSchmiedmayer) [inline TemplateApplication/TemplateApplicationStandard.swift:145]: I think this is a great behavior if we have disable the Firebase interaction. I think the Firebase behaviour should be uploading the PDF to firebase in e.g. the firebase cloud storage: https://firebase.google.com/docs/storage

**Review by PSchmiedmayer — CHANGES_REQUESTED** [review]
Thank you for the great additions, looking forward to see this PR being merged!

I had a small comment regarding the functionality to upload the consent document to Firebase Cloud Storage. Apart from this the PR looks great and the code is well structured 🚀

**Comment by philippzagar** [comment]
Note to ourselves: We need to merge https://github.com/StanfordSpezi/SpeziFirebase/pull/17 first and adjust the dependencies within this branch

File: TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:180]: Just a conversation to remind us that we have to remove that before merging the PR.

File: TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved
Comment (PSchmiedmayer) [inline TemplateApplication.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:207]: Just a conversation to remind us that we have to remove that before merging the PR.

**Review by PSchmiedmayer — APPROVED** [review]
Nice, thank you for the changes; looks great and happy to merge it once we have the related PRs merged and ready 👍

**Pull request merged**
